lib/data_hygiene/govspeak_link_validator.rb

```diff
@@ -12,20 +12,39 @@ def errors
 
         fix = if link.first == "/"
                 unless self.class.is_internal_admin_link?(link)
-                  "Please use either absolute paths for documents created in publisher, e.g. /government/admin/publications/3373, or full URLs for other GOV.UK links"
+                  "Please use either absolute paths for documents created in Whitehall, e.g. /government/admin/publications/3373, or full URLs for other GOV.UK links"
                 end
               elsif self.class.is_internal_admin_link?("/#{link}")
                 "This is an invalid admin link.  Did you mean /#{link} instead of #{link}?"
               elsif !%r{^(?:https?://|mailto:|#)}.match?(link)
                 "Non-document or external links should start with http://, https://, mailto:, or # (for linking to sections on the same page, eg #actions on a policy)"
               elsif link.match?(/whitehall-admin/)
-                "This links to the whitehall-admin domain. Please use paths, eg /government/admin/publications/3373, for documents created in publisher (see guidance on creating links) or full URLs for other GOV.UK links."
+                "This links to the whitehall-admin domain. Please use paths, eg /government/admin/publications/3373, for documents created in Whitehall (see guidance on creating links) or full URLs for other GOV.UK links."
               end
 
         { link:, start: match.begin(0), end: match.end(0), fix: } if fix
       end
     end
 
+    def errors_to_html
```

Comment:
Honestly, i don't really like this approach, but it seemed like the quickest way to get a draft PR open which highlights and resolves the issue.

My preferred approach would be to move a lot of this logic into a view component which handles rendering the sidebar notices (all the inset text components)

The issue with this is that the view component would have to infer when the failure reason was due to link violations and then do a bunch of work to munge that error reason into html.

Which might be fine but seems a little brittle. I guess the edition service object coukd have a constant like 

```
LINK_VIOLATION_POLICY_ERROR_MESSAGE = "This edition contains links which violate linking guidelines".freeze
```

Then the view component could use that constant to determine what action to take.

All the solutions feel a bit sub-optimal though tbh.

Reply:
The problem with moving failure reasons into enums would be that the error message is often constructed dynamically based on the errors on the edition.

https://github.com/alphagov/whitehall/blob/82f2d884acb78bed08bae1568c5e1e8ca17663be/app/services/edition_scheduler.rb#L12-L23

Moving them into their own objects would work though.

A refactor like that's probably beyond the scope of what i can achieve on 2ndline though.

Would you like to to add a view component in this PR, or hand it over to Publishing to put into the backlog? 

It's a real pain for users when they encounter it, but as far as i'm aware it's not a particularly frequent problem.

Reply:
I've carded up something and moved it to the top of ideas https://trello.com/c/pNiIMunP/2224-users-are-given-no-help-identifying-inks-violate-our-policy-in-the-body-for-editions

I'll close this PR for now but it's linked to the Trello ticket so it can be referenced